atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java

```diff
@@ -208,6 +218,28 @@ private static void login(Client client, CassandraCredentialsConfig config) thro
         client.login(new AuthenticationRequest(credsMap));
     }
 
+    /**
+     * Verifies that the current SSL connection hostname/ip address matches what the certificate has served.
+     * This will check both ip address/hostname, and perform (reverse) dns lookups respectively, if hostname
+     * or ip address are not present when supplied.
+     */
+    @VisibleForTesting
+    static void verifyEndpoint(CassandraServer cassandraServer, SSLSocket socket, boolean throwOnFailure)
+            throws SafeSSLPeerUnverifiedException {
+        Set<String> validAddresses =
+                Set.of(socket.getInetAddress().getHostAddress(), cassandraServer.cassandraHostName());
+        boolean endpointVerified =
+                StreamEx.of(validAddresses).anyMatch(address -> hostnameVerifier.verify(address, socket.getSession()));
```

Comment:
nit: we should add to PR description why we chose to write the check this way :)